### ISSUE TITLE
Bitsearch would elevate the content quality to new dimensions 

### ISSUE THREAD

**Issue by ChakraFusion** [TheBeastLT/torrentio-scraper#174]
Hello developer(s) of torrentino,

I very much love your add-on and I am using it almost every day. In most cases the catalogued sources are high quality enough. I am regularly looking for Dolby Vision content in 4k. I am relatively new to real-debrid but I noticed that cocoscrapers on Kodi (Fen) almost always finds higher quality torrents than torrentio (also considering the uncached ones from torrentio).
The latest example of this was with the animated movie "Deep Sea" (2023) where Cocoscrapers found a 9gb DoVi and a 16gb Blu-ray version while torrentio only found some 2gb 1080p versions of the movie. The main difference there was that Cocoscrapers also includes bitsearch.to while torrentio doesn't. 
This lack of bitsearch very apparent with all content since often 50% of all links are fetched from bitsearch. Bitsearch also constantly delivers higher quality movies in every single instance I have checked.
It takes away a lot of the user experience to have to download the Bitsearch movies to RD in Kodi first and then having to wait for it to finish before being able to watch it in Stremio.
Is there a reason why bitsearch isn't included in the torrentio list?
It would elevate the content quality to an enormous degree if it was included. With bitsearch included Torrentio would overtake Kodi in the last missing aspect as well: content quality.

Sincerely Chakra

**Comment by TheBeastLT** [comment]
There were higher resolution sources for this specific movie, but using kitsu catalog since this is anime and there was no mapping from kitsu to imdb.
I'm not planning adding any DHT crawlers atm for reason outlined several times before - https://github.com/TheBeastLT/torrentio-scraper/issues/127

**Comment by ChakraFusion** [comment]
Is there anything I can do to add this kitsu catalog to Torrentio?

From what I understand the reason for not adding DHT crawlers seems to be the bad categorisation. Why does Cocoscrapers get away with the data bitsearch provides? 
Can I do something that would make the implementation of bitsearch possible? I only have limited experience with Java, advanced VBA and some AHK.

**Comment by TheBeastLT** [comment]
> Is there anything I can do to add this kitsu catalog to Torrentio?

There is an addon called `Anime Kitsu`

> Why does Cocoscrapers get away with the data bitsearch provides?

It works in a bit of a different way. Torrentio indexes all torrents it scrapes, and most kodi addons search in the providers based on a keyword. Torrentio needs to assign some imdb id to each torrent it finds, and when categorization is bad sometimes you have non movie titles (porn especially) in movies categories and then it might assign some movie id that looks similar.

> Can I do something that would make the implementation of bitsearch possible? I only have limited experience with Java, advanced VBA and some AHK.

This is beyond scope of this issue, but I'd recommend going through https://github.com/Stremio/stremio-addon-sdk and see if your skill would allow you to create an addon for this.

**Comment by ChakraFusion** [comment]
Thank you for your explaination.
I tried around for a bit and wanted to try what would happen if I manually added the "udp://tracker.bitsearch.to:1337/announce" as provider in magnetHelper.js and in filter.js. Basically to see how bad the miscategorization issue would be. Sadly I got stuck at not being able to run the addon locally. Similar to issue #113.

The whole server hosting stuff is unfamiliar to me and my current skill level doesn't allow me to experiment further.

Would it be possible to add bitsearch as an optional provider (similar to the foreign language providers) to torrentio [maybe with an extra porn warning?] and therefore let only people who don't care for some accidental porn results enable it?
